### DISCUSSION TITLE
Restore Account

### DISCUSSION THREAD

**Discussion by guzuraFe** [KRTirtho/spotube#497] [Q&A]
Device: Android 11, Rooted

I started using spotube few months ago. Few days ago, i was trying to login from pc and discovered i lost my credentials (both mail and password, basically made a new account to test app) to the account. But my app is always updated and I'm still logged in to my account from phone.

Is there any way to recover my account? (sp cookie value/ login credential ) I have root access and I also used browser login AFAIK.

**Comment by KRTirtho** [comment]
Since you've root access you can find the cookies at `/data/data/oss.krtirtho.spotube/app_flutter/spotube_cache.hive`
I'm not exactly sure about the path but it should somewhat similar. Then open the file and find the string `authentication`, next to it you'll see `cookie` where you'll see `sp_dc=<long-value>; sp_key=<short-value>`. Copy them til you see the `accessToken` string. Remove any weird UTF-8 charachters from the strings
Go to: `accounts.spotify.com`. Make sure you're logged out of any other account. Paste these cookies in their respective field. Then reset your email and password

**Reply by guzuraFe** [reply]
Can you suggest how to 'log in from browser' ? Cookies are extracted and tested on a device (from spotube), they are working... But I'm afraid, trying something stupid might result in password reset request by spotify, which will block me out entirely.

When I go to spotify, (with no prior account logged in), i dont see the `sp_dc`, `sp_key` fields.

**Reply by KRTirtho** [reply]
Install the "Cookie Editor" browser extension and go to accounts.spotify.com with no account logged in. Open cookie editor and find the `sp_dc` and `sp_key` fields (If it's not there create them) and replace all the values with your respective cookies. Refresh the page and you should be logged in
